contracts/contracts/VkRegistry.sol

```diff
@@ -11,17 +11,23 @@ import { DomainObjs } from "./utilities/DomainObjs.sol";
 /// Each circuit has a signature which is its compile-time constants represented
 /// as a uint256.
 contract VkRegistry is Ownable(msg.sender), DomainObjs, SnarkCommon, IVkRegistry {
-  mapping(Mode => mapping(uint256 => VerifyingKey)) internal processVks;
-  mapping(Mode => mapping(uint256 => bool)) internal processVkSet;
+  mapping(Mode => mapping(uint256 => VerifyingKey)) public processVks;
+  mapping(Mode => mapping(uint256 => bool)) public processVkSet;
 
   mapping(Mode => mapping(uint256 => VerifyingKey)) internal tallyVks;
   mapping(Mode => mapping(uint256 => bool)) internal tallyVkSet;
 
+  mapping(Mode => mapping(uint256 => VerifyingKey)) internal pollVks;
+  mapping(Mode => mapping(uint256 => bool)) internal pollVkSet;
+
+  event PollVkSet(uint256 _sig, Mode _mode);
```

Comment:
maybe there can be a separate function to set the pollVk? so we avoid setting it twice in that function

Reply:
Do you think it's a better option (cheaper), to have separate functions and two contract calls (setVerifyingKeys & setPollVkKey) or store poll keys per mode? 

I think it's better to have the same approach for all (process, tally, poll) keys even if the poll doesn't depend on the mode
(also code is prettier and easier to read)

As far as I can remember, you told that non-qv will be removed in the future, so then we can remove mode logic, am I right?

Reply:
> you told that non-qv will be removed in the future

mm no the non-qv is not going to be removed in the future.

> to have separate functions and two contract calls (setVerifyingKeys & setPollVkKey) or store poll keys per mode

I think if we can't have all in one, then we should have two as it would be innacurate to store the mode for the poll key when it doesn't have one.

Reply:
great job thank you